### PR TITLE
Add execution of dokken tests as GitHub action

### PR DESCRIPTION
**Pull request by enrico-usai**
### Description of changes
* GitHub will automatically execute the tests specified in the kitchen.recipes.yml as part of the actions executed for each push.
* This permits to ensure all our changes and new tests added are compatible with existing OSes.

### Checklist
- [x] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [x] Check all commits' messages are clear, describing what and why vs how.
- [x] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [x] Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by francesco-giordano — DISMISSED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -85,3 +85,29 @@ jobs:
...
           ignore_paths: third-party
+  dokken-recipes:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        os:
+          - alinux2
+          - centos7
+          - ubuntu18
+          - ubuntu20
```
Comment (awslitvit) [inline .github/workflows/ci.yml:96]: Should we also add `redhat8`?

**Review by francesco-giordano — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (enrico-usai) [reply .github/workflows/ci.yml:96]: yes, but probably is not so easy, so I'll do it later (if I have time). For now at least we're sure we're not breaking existing OSes :)

Comment (awslitvit) [reply .github/workflows/ci.yml:96]: I surely miss something: why shouldn't it be easy, given the set of tests we run? What is the hard part?

Comment (enrico-usai) [reply .github/workflows/ci.yml:96]: The hard part is that the redhat8 image was not available on dokken (locally we're building it). Anyway I found the right base image to use, see patch: https://github.com/aws/aws-parallelcluster-cookbook/pull/1647

**Pull request merged**